tools/tdbg/commands.go

```diff
@@ -371,21 +371,51 @@ func AdminGetShardID(c *cli.Context) error {
 	return nil
 }
 
+// getCategory first searches the registry for the category by the [tasks.Category.Name]. If not found, it checks
+// against a hard-coded mapping of name to category which was defined back when the TaskCategory was an enum. It used to
+// be an enum because we didn't support dynamically-registered task categories, but now we do, so it needs to be an int
+// instead of an enum. It used to look like this:
+//
+//	var TaskCategory_value = map[string]int32{
+//		 "Unspecified": 0,
+//		 "Transfer":    1,
+//		 "Timer":       2,
+//		 "Replication": 3,
+//		 "Visibility":  4,
+//		 "Archival":    5,
+//		 "MemoryTimer": 6,
+//	}
+func getCategory(registry tasks.TaskCategoryRegistry, key string) (tasks.Category, error) {
+	for _, category := range registry.GetCategories() {
+		if category.Name() == key {
```

Comment:
nit: shall we make the comparison case insensitive (and also update the desc of the flag)?

Reply:
Made it case insensitive and updated the description.